src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -2137,6 +2137,25 @@ public void GenerateMethod(Method method, Class @class)
 
         private void GenerateOverloadCall(Function function)
         {
+            var needUnsafe = function.Parameters.Any(p => p.Kind == ParameterKind.Regular && p.Ignore
+                             && p.Type.IsPointerToPrimitiveType() && p.Usage == ParameterUsage.InOut && p.HasDefaultValue);
+            if(needUnsafe)
+            {
+                WriteLine("unsafe");
```

Comment:
This does work but please move it to the method signature instead, it's less cumbersome.